mlte/spec/spec.py

```diff
@@ -35,200 +36,131 @@ def _unique(collection: List[str]) -> bool:
 # -----------------------------------------------------------------------------
 
 
-class Spec:
+class Spec(Artifact):
     """
     The Spec class integrates properties, measurements,
     and the results of measurement evaluation and validation.
     """
 
     def __init__(
-        self, requirements_by_property: Dict[Property, List[Requirement]]
+        self, identifier: str, properties: Dict[Property, Dict[str, Condition]]
     ):
         """
         Initialize a Spec instance.
 
-        :param properties: The collection of properties that compose the spec.
+        :param properties: The collection of properties that compose the spec, with their conditions keyed by measurement id.
         :type properties: List[Property]
         """
-        self.properties = [p for p in requirements_by_property.keys()]
+        super().__init__(identifier, ArtifactType.SPEC)
+
+        self.properties = properties
         """The collection of properties that compose the Spec."""
 
-        if not _unique([p.name for p in self.properties]):
+        if not _unique([p.name for p in self.properties.keys()]):
             raise RuntimeError("All properties in Spec must be unique.")
 
         if not _unique(
-            [
-                str(requirement.identifier)
-                for _, req_list in requirements_by_property.items()
-                for requirement in req_list
-            ]
+            list(
```

Comment:
This is a beautiful expression, but is this true? Should it not be possible to utilize the same "evidence" to validate more than one property? I know this is something of a philosophical question, but I don't see an immediate reason to prevent this.

Reply:
That is a good question. This isn't something I just added, btw, it was there from the previous implementation with Requirements, just in a slightly different form, but the question is still valid. Right now, there is an implementation issue with removing this. Not in the Spec itself, but when validating the Spec. In the `SpecValidator`, results are stored by measurement_id, but not separately by property, which means that two conditions using the same `Value` (with the same measurement id) would end up overwriting one another in the list of results. This is more of a limitation of the SpecValidator than a high level requirement, though. I can look into this, since I think refactoring SpecValidator a bit to support this should be simple, but I would prefer to do it in the subsequent PR, #237, which 1) deals more with the validation aspects, and 2) is based on this PR.